darshan-util/pydarshan/darshan/backend/cffi_backend.py

```diff
@@ -300,13 +300,15 @@ def log_get_record(log, mod, dtype='numpy'):
 
 
 
-def log_get_generic_record(log, mod_name, dtype='numpy'):
+def log_get_generic_record(log, mod_name, c_cols=None, fc_cols=None, dtype='numpy'):
```

Comment:
Generating the dictionary keys/dataframe column names for each record was consuming a lot of run time (I think it was ~30%), even after most of the optimizations were implemented, so I added the (optional) arguments to input them. 

I originally thought this would be dangerous because the key/column names might change for different records, but the default `numpy` output doesn't use these anyways (it uses the same key/column names as input in `mod_read_all_records()`). So it is implicitly assumed that these are the same when using the default.

Reply:
Actually, this change isn't fully safe, as I noted above.

Reply:
(though these functions are likely largely "private," but still best to follow backwards-compatibility conventions instead of hard-breaking possible uses)

Reply:
I'll resolve this since these changes have been removed.